jdbc/src/main/java/nextstep/jdbc/JdbcTemplate.java

```diff
@@ -1,17 +1,76 @@
 package nextstep.jdbc;
 
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.sql.DataSource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.sql.DataSource;
-
 public class JdbcTemplate {
 
     private static final Logger log = LoggerFactory.getLogger(JdbcTemplate.class);
+    private static final int SINGLE_RESULT_SIZE = 1;
 
     private final DataSource dataSource;
 
     public JdbcTemplate(final DataSource dataSource) {
         this.dataSource = dataSource;
     }
+
+    public void update(final String sql, final Object... args) {
+        try (Connection conn = dataSource.getConnection();
+             PreparedStatement pstmt = conn.prepareStatement(sql)) {
+            setArguments(pstmt, args);
+            pstmt.executeUpdate();
+        } catch (SQLException e) {
+            log.error(e.getMessage(), e);
+            throw new DataAccessException(e);
+        }
+    }
+
+    public <T> T queryForObject(final String sql, final RowMapper<T> rowMapper, final Object... args) {
+        List<T> results = query(sql, rowMapper, args);
+        return getSingleResult(results);
+    }
+
+    public <T> List<T> query(final String sql, final RowMapper<T> rowMapper, final Object... args) {
+        try (Connection conn = dataSource.getConnection();
+             PreparedStatement pstmt = conn.prepareStatement(sql)) {
+            setArguments(pstmt, args);
+            ResultSet rs = pstmt.executeQuery();
+
+            return getResult(rowMapper, rs);
+        } catch (SQLException e) {
+            log.error(e.getMessage(), e);
+            throw new DataAccessException(e);
+        }
+    }
+
+    private <T> T getSingleResult(final List<T> results) {
+        if (results.size() != SINGLE_RESULT_SIZE) {
+            throw new DataAccessException("조회한 결과의 크기가 1이 아닙니다.");
+        }
+        return results.get(0);
+    }
+
+    private <T> List<T> getResult(final RowMapper<T> rowMapper, final ResultSet rs) throws SQLException {
+        List<T> results = new ArrayList<>();
+        while (rs.next()) {
+            results.add(rowMapper.mapRow(rs, rs.getRow()));
+        }
+        rs.close();
+        return results;
+    }
```

Comment:
```suggestion
    private <T> List<T> getResult(final PreparedStatement preparedStatement,
                                  final RowMapper<T> rowMapper) throws SQLException {
        try (ResultSet rs = preparedStatement.executeQuery()) {
            List<T> result = new ArrayList<>();
            while (rs.next()) {
                result.add(rowMapper.mapRow(rs, rs.getRow()));
            }
            return result;
        }
    }
```
제가 의도한 `ResultSet`관련 코멘트는 메서드 분리를 통해 `try-resource-finally`를 하는 것은 위와 같이 수정하는 것이었어요!! 

`ResultSet`을 만들어 넘겨주기보다 호출한 메서드에서 `ResultSet`을 만들게 된다면 `try-resource-finally`를 사용할 수 있을 것 같아요👍